### PR TITLE
feat(report-issue-screen): Report Issue Screen in Developer Menu

### PR DESCRIPTION
**Pull request by thisisabhash**
*Issue #, if available:*
None 

*Description of changes:*
1. Report issue screen in developer menu
2. Added code to fetch environment info from json file
3. Added code to wrap customer supplied LoggingCategoryPlugin inside PersistentLoggingPlugin

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Amplify/DevMenu/Data/EnvironmentInfoHelper.swift [hunk 1]
```diff
@@ -12,6 +12,25 @@ import UIKit
 @available(iOS 13.0.0, *)
 struct EnvironmentInfoHelper {
 
+    static let environmentInfoSourceFileName = "local-env-info"
+
+    static func fetchDeveloperInformationFromJson(filename: String) -> [EnvironmentInfoItem] {
+        guard let url = Bundle.main.url(forResource: filename, withExtension: "json") else {
+            print("Error : json file doesn't exist")
```
Comment (royjit) [inline Amplify/DevMenu/Data/EnvironmentInfoHelper.swift:19]: Use Amplify logger to print out a warning.

File: Amplify/DevMenu/Data/EnvironmentInfoHelper.swift [hunk 2]
```diff
@@ -12,6 +12,25 @@ import UIKit
...
+            return [EnvironmentInfoItem]()
+        }
+
+        do {
+            let jsonData = try Data(contentsOf: url)
+            let decoder = JSONDecoder()
+            let environmentInfo = try decoder.decode(DevEnvironmentInfo.self, from: jsonData)
+            return getDeveloperEnvironmentInformation(devEnvInfo: environmentInfo)
+        } catch {
+            print("Error : json file parsing failed")
```
Comment (royjit) [inline Amplify/DevMenu/Data/EnvironmentInfoHelper.swift:29]: Use Amplify Logger

File: Amplify/DevMenu/View/IssueReporter.swift [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+        let issueDescriptionMarkdown =
+            IssueInfoHelper.generateMarkdownForIssue(
+                issue: IssueInfo(issueDescription: issueDescription,
+                         includeEnvInfo: includeEnvInfo,
+                         includeDeviceInfo: includeDeviceInfo,
+                         includeLogs: includeLogs))
+
+        let urlString = amplifyIosNewIssueUrl + issueDescriptionMarkdown
+        guard let url = urlString.addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed) else { return }
+        guard let urlToOpen = URL(string: url) else { return }
```
Comment (royjit) [inline Amplify/DevMenu/View/IssueReporter.swift:84]: Show some alert to the user if they cannot move forward

File: Amplify/DevMenu/AmplifyDevMenu.swift [hunk 4]
```diff
@@ -31,7 +31,8 @@ public final class AmplifyDevMenu: DevMenuBehavior, TriggerDelegate {
     public func showMenu() {
         guard let rootViewController =
             devMenuPresentationContextProvider?.devMenuPresentationContext().rootViewController else {
-                print("Warning: RootViewController of the UIWindow is nil")
+                Amplify.Logging.warn(DevMenuStringConstants.logTag +
+                    "Warning: RootViewController of the UIWindow is nil")
```
Comment (royjit) [inline Amplify/DevMenu/AmplifyDevMenu.swift:35]: Remove = `Warning: `

File: Amplify/DevMenu/Data/EnvironmentInfoHelper.swift [hunk 5]
```diff
@@ -12,6 +12,25 @@ import UIKit
 @available(iOS 13.0.0, *)
 struct EnvironmentInfoHelper {
 
+    static let environmentInfoSourceFileName = "local-env-info"
+
+    static func fetchDeveloperInformationFromJson(filename: String) -> [EnvironmentInfoItem] {
+        guard let url = Bundle.main.url(forResource: filename, withExtension: "json") else {
+            Amplify.Logging.error(DevMenuStringConstants.logTag + "Error : json file doesn't exist")
```
Comment (royjit) [inline Amplify/DevMenu/Data/EnvironmentInfoHelper.swift:19]: Here and in other places remove - `Error :`

File: Amplify/DevMenu/Data/IssueInfo.swift [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+             }
+         }
+
+         return description
+     }
+
+    private func getItemsDescription(items: [InfoItemProvider]) -> String {
+        var description: String
+
+        if items.isEmpty {
```
Comment (royjit) [inline Amplify/DevMenu/Data/IssueInfo.swift:103]: Maybe change the code like this:
``` swift
guard !items.isEmpty else {
   return infoNotAvailable
} 
return items.reduce("") {(description, item) -> String in
    return description.append("\(item.displayName) - \(item.information) \n")
}

```

**Comment by thisisabhash** [comment]
Screenshots on different devices (in portrait and landscape mode)

https://user-images.githubusercontent.com/23294582/88701183-502aa080-d0cf-11ea-8241-c0816101a02c.png
https://user-images.githubusercontent.com/23294582/88701187-50c33700-d0cf-11ea-88de-b38f02d9b69c.png
https://user-images.githubusercontent.com/23294582/88701190-515bcd80-d0cf-11ea-9e3c-cd093540f794.png
https://user-images.githubusercontent.com/23294582/88701199-51f46400-d0cf-11ea-89c6-52b1cff39039.png
https://user-images.githubusercontent.com/23294582/88701200-51f46400-d0cf-11ea-836b-011f93db2ce3.png
https://user-images.githubusercontent.com/23294582/88701203-528cfa80-d0cf-11ea-8808-2e665740a35c.png
https://user-images.githubusercontent.com/23294582/88701204-53259100-d0cf-11ea-8a99-7307bf8e89e2.png
https://user-images.githubusercontent.com/23294582/88701207-53be2780-d0cf-11ea-8b59-39cf7e58213e.png
https://user-images.githubusercontent.com/23294582/88701208-53be2780-d0cf-11ea-9593-307303620f94.png
https://user-images.githubusercontent.com/23294582/88701212-5456be00-d0cf-11ea-8985-d5e9fa65ea7a.png
https://user-images.githubusercontent.com/23294582/88701216-5587eb00-d0cf-11ea-815d-b4331c68d5d2.png
https://user-images.githubusercontent.com/23294582/88701219-5587eb00-d0cf-11ea-98be-13be47a4b854.png

**Pull request merged**